wakuv2/waku.go

```diff
@@ -1012,11 +1012,22 @@ func (w *Waku) publishEnvelope(envelope *protocol.Envelope, publishFn publishFn,
 	defer w.wg.Done()
 
 	var event common.EventType
-	if err := publishFn(envelope, logger); err != nil {
-		logger.Error("could not send message", zap.Error(err))
-		event = common.EventEnvelopeExpired
-	} else {
-		event = common.EventEnvelopeSent
+	for {
+		err := publishFn(envelope, logger)
+		if err != nil {
+			if err.Error() == "not enough peers to publish" {
+				logger.Warn("Not enough peers to publish, will retry after 3 seconds", zap.Stringer("envelopeHash", envelope.Hash()))
+				time.Sleep(3 * time.Second)
+				continue // Retry publishing
+			} else {
+				logger.Error("could not send message", zap.Error(err))
+				event = common.EventEnvelopeExpired
+				break // Exit loop on other errors
+			}
+		} else {
+			event = common.EventEnvelopeSent
+			break // Exit loop on successful publish
+		}
```

Comment:
The previous `if` either `continue` or `break` the loop, so it shouldn't be necessary to use an `else` here.
```suggestion
		event = common.EventEnvelopeSent
		break // Exit loop on successful publish
```